### DISCUSSION TITLE
Bank account and carried money

### DISCUSSION THREAD

**Discussion by AbsDataStruct** [nikosgram/gringotts#79] [Ideas]
I've searched for long to get rid of wire transfer ... forcing player to carry their money in their inventory in order to use it !
Trying to make enhancement to the ChestShop plugin for such item money system, they let me know about the Gringotts plugin which is exactly what I wanted !!! Thanks for this great plugin !

Some other ideas or suggestions :
1) allow player to create vault only in some Gringotts Banks area (using worldguard plugin region)
2) set restriction on /money deposit and /money withdraw to be used only near the vaults
3) create ATM sign that would enable to remotely withdraw or deposit (when needed) with a transaction fee (and a NPC running to bring the money items)
4) If possible, the /pay to pay another player should only take money items from the player's inventory and not the vault, and give money items to the player's inventory as well

**Comment by nikosgram** [comment]
All of them are actually really good ideas. I recently removed support for worldguard as nobody was using it. Though, the support was based on the region. It would be easy to implement it back though in the way you suggested.

About the 2,3,4 suggestions though, they are amazing, but, they would change the way gringotts works, and that would upset lot of people. I could work on a different kind of economy plugin that would provide that kind of features, but, it would be nothing like Gringotts. Also, no promises here :P

**Comment by AbsDataStruct** [comment]
Thanks for your answer !

You removed what you made related to worldguard ... is it not possible to toggle the use of worldguard in the config ? I don't think it would make any performance issue, and the plugin would be still usable like before ...
What do you think ?

About 1) The Goal is not only to force player to carry money from their house to the shop in order to buy something but also force vendor to retrieve their money from a bank and carry it at home ... Since the vendor could receive payment to his Gringotts account.
For that, we would restrict the vault's position in some Gringotts banks areas ...

About 2) It is not really usefull, since commands altogether could be simply removed, commands are annoying anyway ... :-)
We would force the player to go to the vault itself...

About 3) Personally, I would not use ATM, another plugin could do the items transports/delivery system with a NPC ...
Electronic money is what I want to avoid, but maybe some server would like to have that possibility anyway and not restrict to items too much...

About 4) I'm not sure it is already the case ... If it is already the case when players are connected (use both player's inventories), that's good enough for me ... of course, making a payment when player are offline should use the vault ... or maybe, enabling config option to allow payment from/to offline player or not and restrict the pay command to only online player ... Is that not too difficult to do ?

Thanks for your help ...
I'm dreaming of item only economy, without remote exchange, withdraw or payment, since I started my minecraft server in version 1.2 !! 
The user should manipulate the money item and see it in their hands all the time ... Minecraft is concrete and visual game, money is quite abstract for small children ... I would love such economy so much ... sure children would too !
Moreover, it contributes to force exchanges between other players ... I don't like player staying alone or by small group in their plot, and really wish to use 
I can develop in Java, and I would like to make a try to help if needed ...
;-)
